server/app/watchlist.py

```diff
@@ -56,6 +57,53 @@ def __iter__(self):
 	def is_invalid(self):
 		return self.__is_invalid
 
+	def write(self):
+		self.write_watchlist(self, self.path)
+
+	def add_channel(self, id=None, source='youtube', site=None, name='channel'):
+		if id is None:
+			raise TypeError('Missing keyword argument "id", string expected')
+		source = source.lower()
+		if source not in url_formats:
+			raise ValueError('source not in url formats')
+		if site is None:
+			site = url_formats[source]['default_site']
+		self.channels.append({
+			'name': name,
```

Comment:
I don't think we should accept custom names for channels, in the UI we'll just show the name from the feed. Let's just store ids (and other metadata like source) in the file.

Reply:
I should have used the name 'label' instead of 'name'.
The 'name' field is not actually meant to be a custom name for the channel.
It is only used as the section label in the watchlist ini file, and does not affect the output of route /videos in anyway.

The field is mainly there to improve readability when adding entries to the watchlist file, but I haven't implemented
a way to make the default name the channel name yet.

Reply:
I still don't like it. I see that you use the name to mutate the config, which can easily lead to bugs as channel names aren't unique. Also you default `name` to "channel", which will definitely result in bugs because front end will only send the id/url and hence there'll be several channels with name "channel" and they will overwrite each other.
Let's just use ids and forget about labels. The file being human readable can be a plus but not a high priority.

Reply:
Section labels are mandatory for ini files.
It shouldn't be an issue even if all entries have the same label though, since the code will only match
an entry that matches all of the conditions, any unspecified condition will be as if that entry isn't there in the first place.

Since 'id' is the only necessary field and it's mandatory, there shouldn't be any problems with the labels.

Reply:
Let's just use id as a label, it's failproof this way